### PR TITLE
Update scala dependencies

### PR DESCRIPTION
**Pull request by seblm**
- junit from 4.12 to 5.7.0
 - sbt from 0.13.8 to 1.5.0
 - scala from 2.11.7 to 2.13.5
 - slf4j-simple from 1.7.13 to 1.7.30

Also move tests to a package and remove useless production code.

I am not sure about reformatting scala code with scalafmt because you don't want any pull request reformat a file. Please let me know if you want me to revert reformatting.

Thank you for delivering JGiven: it's a great tool!

### CODE REVIEW

**Review by l-1squared — COMMENTED** [review]
Hi @seblm, 
first off, many thanks for your contriubution. I greatly appreciate your effort in improving the scala example project, especially because my expertise regarding scala is rather limited. 

Besides the comments i left in this review, I would like to ask you to rename your commit. While formating and moving the files within this commit is fine by me, I think such action should be reflected in the commit message.  Maybe a message like "Refurbish scala example project" would be more fitting

**Comment by l-1squared** [comment]
@l-1squared  at merge: add contribution by saiblerm

File: example-projects/scala/src/test/scala/com/tngtech/jgiven/scala/example/SimpleHelloTest.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.junit.jupiter.api.Test
+
+class SimpleHelloTest extends SimpleScenarioTest[Steps] {
+
+  @Test
+  def my_first_JGiven_scenario_in_Scala(): Unit = {
+    given.some_another_state()
+    when.some_action()
+    `then`.some_outcome
+    ()
```
[the remark this answers is not in the document]
Comment (seblm) [reply example-projects/scala/src/test/scala/com/tngtech/jgiven/scala/example/SimpleHelloTest.scala:13]: You've right : there is no need to explicitly terminates each test with unit value. I'll remove that.

File: example-projects/scala/project/build.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #Activator-generated Properties
```
[the remark this answers is not in the document]
Comment (seblm) [reply example-projects/scala/project/build.properties]: That's a good idea.

**Comment by seblm** [comment]
> @l-1squared at merge: add contribution by saiblerm

Maybe _saiblerm_ had some typos. Do you mean _seblm_?

**Comment by seblm** [comment]
> Hi @seblm,
> first off, many thanks for your contriubution. I greatly appreciate your effort in improving the scala example project, especially because my expertise regarding scala is rather limited.
> 
> Besides the comments i left in this review, I would like to ask you to rename your commit. While formating and moving the files within this commit is fine by me, I think such action should be reflected in the commit message. Maybe a message like "Refurbish scala example project" would be more fitting

@l-1squared, I have rename the commit accordingly.

**Comment by l-1squared** [comment]
> > @l-1squared at merge: add contribution by saiblerm
> 
> Maybe _saiblerm_ had some typos. Do you mean _seblm_?

yes :)

**Review by l-1squared — APPROVED** [review]

**Pull request merged**